src/GovUk.Education.ExploreEducationStatistics.Admin/Services/PreReleaseUserService.cs

```diff
@@ -153,6 +158,93 @@ public class PreReleaseUserService : IPreReleaseUserService
+                                .Include(r => r.User)
+                                .Where(
+                                    r =>
+                                        r.ReleaseId == releaseId
+                                        && r.User.Email.ToLower() == email.ToLower()
+                                        && r.Role == ReleaseRole.PrereleaseViewer
+                                )
+                        );
+                        _context.RemoveRange(
+                            _context
+                                .UserReleaseInvites
+                                .Where(
+                                    i =>
+                                        i.ReleaseId == releaseId
+                                        && i.Email.ToLower() == email.ToLower()
+                                        && i.Role == ReleaseRole.PrereleaseViewer
+                                )
+                        );
+                        await _context.SaveChangesAsync();
+
+                        var remainingReleaseInvites = await _context
+                            .UserReleaseInvites
+                            .Where(
+                                i =>
+                                    i.Email.ToLower() == email.ToLower()
+                                    && i.Role == ReleaseRole.PrereleaseViewer
+                            )
+                            .CountAsync();
+
+                        if (remainingReleaseInvites == 0)
+                        {
+                            var role = await GetPreReleaseUserRole();
+
+                            _usersAndRolesDbContext.UserInvites.RemoveRange(
+                                _usersAndRolesDbContext.UserInvites
+                                    .Where(
+                                        i =>
+                                            i.Email.ToLower() == email.ToLower()
+                                            && i.RoleId == role.Id
+                                            && !i.Accepted
+                                    )
+                            );
+
+                            await _usersAndRolesDbContext.SaveChangesAsync();
+                        }
+                    }
+                );
+        }
+
+        public async Task<Either<ActionResult, Unit>> SendPreReleaseUserInviteEmails(Release release)
+        {
+            var userReleaseInvites = await _context.UserReleaseInvites
+                .Where(i =>
+                    i.ReleaseId == release.Id
+                    && i.Role == ReleaseRole.PrereleaseViewer
+                    && i.EmailSent == false)
+                .ToListAsync();
+
+            await userReleaseInvites.ForEachAsync(async invite =>
+            {
+                var user = await _context.Users
+                    .SingleOrDefaultAsync(u => u.Email.ToLower() == invite.Email.ToLower());
+                SendPreReleaseInviteEmail(release, invite.Email.ToLower(), user == null);
```

Comment:
Might be good to use a named parameter on the `user == null` part, to describe what is being achieved i.e. what the flag signifies.